lib/core/src/sync/mod.rs

```diff
@@ -1 +1,189 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+
+use crate::{persist::Persister, prelude::Signer};
+
+use self::client::SyncerClient;
+use self::model::DecryptedRecord;
+use self::model::{
+    data::{ChainSyncData, ReceiveSyncData, SendSyncData, SyncData},
+    sync::ListChangesRequest,
+    RecordType, SyncState,
+};
+
+pub(crate) mod client;
 pub(crate) mod model;
+
+macro_rules! unwrap_or_continue {
+    ($result:expr, $dbg:literal) => {
+        match $result {
+            Ok(val) => val,
+            Err(err) => {
+                log::debug!($dbg, err);
+                continue;
+            }
+        }
+    };
+}
+
+pub(crate) struct SyncService {
+    remote_url: String,
+    persister: Arc<Persister>,
+    signer: Arc<Box<dyn Signer>>,
+    client: Box<dyn SyncerClient>,
+}
+
+impl SyncService {
+    pub(crate) fn new(
+        remote_url: String,
+        persister: Arc<Persister>,
+        signer: Arc<Box<dyn Signer>>,
+        client: Box<dyn SyncerClient>,
+    ) -> Self {
+        Self {
+            remote_url,
+            persister,
+            signer,
+            client,
+        }
+    }
+
+    fn apply_record(&self, decrypted_record: &DecryptedRecord) -> Result<()> {
+        match decrypted_record.data.clone() {
+            SyncData::Chain(chain_data) => self.persister.insert_chain_swap(&chain_data.into())?,
+            SyncData::Send(send_data) => self.persister.insert_send_swap(&send_data.into())?,
+            SyncData::Receive(receive_data) => {
+                self.persister.insert_receive_swap(&receive_data.into())?
+            }
+        }
+        Ok(())
+    }
+
+    fn fetch_sync_data(&self, data_id: &str, record_type: RecordType) -> Result<SyncData> {
```

Comment:
NIT: since it is local fetch (a bit confusing with remote fetch) perhaps we can rename to `load_local_copy` / load_sync_data ?